### PR TITLE
Update settings_backup.php

### PR DESCRIPTION
**Pull request by aftechro**
Create backups and restore from list. TODO: restore from file

### CODE REVIEW

**Comment by wrongecho** [comment]
Test these changes at: https://patch14858.pr-review.itflow.org  
 _(automatic message)_

File: settings_backup.php [hunk 1]
```diff
@@ -1,37 +1,250 @@
...
+
+// Handle backup action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['backup'])) {
+    // Create a backup
+    $backupFileName = date("d-m-Y_H-i-s") . ".sql";
+    $backupPath = $backupFolder . $backupFileName;
+
+    // Run mysqldump command to include table content
+    $escapedBackupPath = escapeshellarg($backupPath);
+    $command = "mysqldump --complete-insert --skip-comments -h $dbhost -u $dbusername -p$dbpassword $database > $escapedBackupPath";
```
[the remark this answers is not in the document]
Comment (aftechro) [reply settings_backup.php:31]: false positive

File: settings_backup.php [hunk 2]
```diff
@@ -1,37 +1,250 @@
...
+    if (isset($_POST['selectedBackups'])) {
+        foreach ($_POST['selectedBackups'] as $selectedBackup) {
+            $backupPath = $backupFolder . $selectedBackup;
+
+            // Validate the file path to prevent directory traversal
+            $realBackupPath = realpath($backupPath);
+            $realBackupFolder = realpath($backupFolder);
+
+            if ($realBackupPath !== false && $realBackupFolder !== false && strpos($realBackupPath, $realBackupFolder) === 0) {
+                unlink($backupPath);
```
[the remark this answers is not in the document]
Comment (aftechro) [reply settings_backup.php:95]: have no idea what else to do here, changed about 5/6 codes now, and still get the same. chatgpt saying is ok

**Comment by aftechro** [comment]
> Test these changes at: https://patch14858.pr-review.itflow.org _(automatic message)_

this can now be seen and tested in demo pr

**Comment by aftechro** [comment]
demo works for doing backups, but i`m getting some errors on restore, not sore if php versioning or mariadb versus mysql server

**Pull request closed without merging**